commit 4aeea489ed3189d7ebcae4199153c1ae736ebf68
Author: Daniel Vetter <daniel.vetter@ffwll.ch>
Date:   Sun Oct 30 16:07:45 2011 +0100

    i810: rip out PIO access functions
    
    All the io done at init time is already converted to libpciaccess
    
    Signed-Off-by: Daniel Vetter <daniel.vetter@ffwll.ch>

diff --git a/src/legacy/i810/i810.h b/src/legacy/i810/i810.h
index 6620695..c3a6255 100644
--- a/src/legacy/i810/i810.h
+++ b/src/legacy/i810/i810.h
@@ -178,7 +178,6 @@ typedef struct _I810Rec {
    int Chipset;
    unsigned long LinearAddr;
    unsigned long MMIOAddr;
-   IOADDRESS ioBase;
    EntityInfoPtr pEnt;
    struct pci_device *PciInfo;
 
diff --git a/src/legacy/i810/i810_driver.c b/src/legacy/i810/i810_driver.c
index 3637e25..83c999f 100644
--- a/src/legacy/i810/i810_driver.c
+++ b/src/legacy/i810/i810_driver.c
@@ -234,7 +234,6 @@ I810DoDDC(ScrnInfoPtr pScrn, int index)
 static Bool
 I810PreInit(ScrnInfoPtr pScrn, int flags)
 {
-   vgaHWPtr hwp;
    I810Ptr pI810;
    ClockRangePtr clockRanges;
    int i;
@@ -270,8 +269,6 @@ I810PreInit(ScrnInfoPtr pScrn, int flags)
    /* Allocate a vgaHWRec */
    if (!vgaHWGetHWRec(pScrn))
       return FALSE;
-   hwp = VGAHWPTR(pScrn);
-   pI810->ioBase = hwp->PIOOffset;
 
    pI810->PciInfo = xf86GetPciInfoForEntity(pI810->pEnt->index);
 
@@ -378,9 +375,6 @@ I810PreInit(ScrnInfoPtr pScrn, int flags)
     */
    I810DoDDC(pScrn, pI810->pEnt->index);
 
-   /* We have to use PIO to probe, because we haven't mapped yet */
-   I810SetPIOAccess(pI810);
-
    intel_detect_chipset(pScrn, pI810->PciInfo, &chipset);
 
    /*
diff --git a/src/legacy/i810/i810_io.c b/src/legacy/i810/i810_io.c
index b485cef..954ed6e 100644
--- a/src/legacy/i810/i810_io.c
+++ b/src/legacy/i810/i810_io.c
@@ -45,43 +45,6 @@ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 #define moutb(p,v) *(volatile uint8_t *)(pI810->MMIOBase + (p)) = (v)
 
 static void
-I810WriteControlPIO(I810Ptr pI810, IOADDRESS addr, uint8_t index, uint8_t val)
-{
-   addr += pI810->ioBase;
-   outb(addr, index);
-   outb(addr + 1, val);
-}
-
-static uint8_t
-I810ReadControlPIO(I810Ptr pI810, IOADDRESS addr, uint8_t index)
-{
-   addr += pI810->ioBase;
-   outb(addr, index);
-   return inb(addr + 1);
-}
-
-static void
-I810WriteStandardPIO(I810Ptr pI810, IOADDRESS addr, uint8_t val)
-{
-   outb(pI810->ioBase + addr, val);
-}
-
-static uint8_t
-I810ReadStandardPIO(I810Ptr pI810, IOADDRESS addr)
-{
-   return inb(pI810->ioBase + addr);
-}
-
-void
-I810SetPIOAccess(I810Ptr pI810)
-{
-   pI810->writeControl = I810WriteControlPIO;
-   pI810->readControl = I810ReadControlPIO;
-   pI810->writeStandard = I810WriteStandardPIO;
-   pI810->readStandard = I810ReadStandardPIO;
-}
-
-static void
 I810WriteControlMMIO(I810Ptr pI810, IOADDRESS addr, uint8_t index, uint8_t val)
 {
    moutb(addr, index);
